widgets/PageListItemTitle.php

```diff
@@ -103,6 +105,25 @@ public function run()
             'level' => $this->level,
         ]);
     }
+    protected function generateNumbering($level)
+    {
```

Comment:
Please add unit tests for logic. See for general reference: https://docs.humhub.org/docs/develop/testing/

Reply:
UPDATE: I have implement the unit test logic for generateNumbering and commited it below